src/node/rpc/memberfrontend.h

```diff
@@ -395,10 +395,25 @@ namespace ccf
         if (verify_result != QuoteVerificationResult::VERIFIED)
           return QuoteVerifier::quote_verification_error_to_json(verify_result);
 #endif
+        auto nodes_view = args.tx.get_view(this->network.nodes);
+        bool is_duplicate = false;
+        // TODO(#api): foreach should check the callback's value and be able to
+        // stop once the returned value is false
+        nodes_view->foreach(
+          [&new_node, &is_duplicate](const NodeId& nid, const NodeInfo& ni) {
+            if (
+              !is_duplicate &&
+              (new_node.tlsport == ni.tlsport && new_node.host == ni.host))
+              is_duplicate = true;
+          });
+        if (is_duplicate)
+          return jsonrpc::error(
+            jsonrpc::ErrorCodes::INVALID_PARAMS,
+            "A node with the same host and port already exists");
```

Comment:
This error message should include the host and port, and ideally the NodeId of the duplicate (so the member knows which node to remove to resolve this).